core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -5055,4 +5055,24 @@ void checkUserDefinedOrderByOver(NullCollation nullCollation) {
     String sql = "SELECT CAST(CAST(? AS INTEGER) AS CHAR)";
     sql(sql).ok();
   }
+
```

Comment:
there is no documentation added in this PR for these behaviors.
Also, it would be nice to have some negative tests, where type inference rejects the query, and with NULL arguments.

Reply:
I'll try to come up with a negative test shortly. In the meantime, I've added documentation for the new overloads, and made some minor corrections to the existing documentation so they more accurately reflect how BigQuery actually handles these functions.

The no-op versions are undocumented by BigQuery, but easily verifiable as working with a simple example query: `SELECT DATETIME(DATETIME '2022-02-02 12:12:12')`